### ISSUE TITLE
Can't use queries with variables in refetchQueries

### ISSUE THREAD

**Issue by MargaretKrutikova** [apollographql/reason-apollo#203]
Currently you can only pass `string` in `refetchQueries` when calling mutations, because the type `apolloMutation` in `ReasonApolloMutation` defines `refetchQueries` as optional `array(string)`. 

However, it seems like in many cases you would actually want to pass variables to those queries, `apollo-client` defines it like this:
```ts
type PureQueryOptions = {
  query: DocumentNode;
  variables?: { [key: string]: any };
  context?: any;
};
type RefetchQueryDescription = Array<string | PureQueryOptions>;
...
refetchQueries?:
    | ((result: ExecutionResult<T>) => RefetchQueryDescription)
    | RefetchQueryDescription;
```

In `reason-apollo`, there is a type `queryObj` in `ApolloClient`:
```ocaml
type queryObj = {
  .
  "query": ReasonApolloTypes.queryString,
  "variables": Js.Json.t,
};
```
so I quickly tried making `refetchQueries` into `unit => array(queryObj)=?` and it allowed me to pass variables into queries in `refetchQueries`. 

However I am not sure how to make it work for both for strings and proper query objects. Is it even possible to do in `reason`? 

- [ ] has-reproduction
- [x] feature
- [ ] docs
- [ ] blocking
- [ ] good first issue

**Comment by Gregoirevda** [comment]
Good point! I guess it would require both types to be wrapped in a variant which can then be deconstructed in ReasonApolloMutation. That would be a breaking change. Could you create a PR?

**Comment by MargaretKrutikova** [comment]
Yes, absolutely! I will prepare a PR.
